python/tank/path_cache.py

```diff
@@ -572,250 +565,417 @@ def _do_full_sync(self, cursor):
         
         return data
 
+    @classmethod
+    def remove_filesystem_location_entries(cls, tk, path_ids):
+        """
+        Removes FilesystemLocation entries from the path cache.
+
+        :param list path_ids: List of FilesystemLocation entries to remove.
```

Comment:
nitpick: FilesystemLocation *entries* sound like they are entity dictionaries. I suggest switching to say FilesystemLocation ids.